### PR TITLE
feat: hpo components

### PR DESCRIPTION
**Pull request by frascuchon**
This PR simplify ray.tune hpo integration

### CODE REVIEW

File: src/biome/text/features.py [hunk 1]
```diff
@@ -61,7 +61,10 @@ def config(self) -> Dict:
 
     def to_json(self) -> Dict:
         """Returns the config as dict for the serialized json config file"""
-        return self.to_dict()
+        data = vars(self)
+        data.update(data.pop("extra_params", {}))
+
+        return data
 
     def to_dict(self) -> Dict:
```
Comment (dcfidalgo) [inline src/biome/text/features.py:69]: This is the same as `to_json`, not sure why we have both methods really ... But i would either let `to_json` return `to_dict` or `to_dict` return `to_json`, not implementing the logic twice.

File: src/biome/text/features.py [hunk 2]
```diff
@@ -140,7 +143,10 @@ def config(self) -> Dict:
 
     def to_json(self):
         """Returns the config as dict for the serialized json config file"""
-        return self.to_dict()
+        data = vars(self)
+        data.update(data.pop("extra_params", {}))
+
+        return data
 
     def to_dict(self):
```
Comment (dcfidalgo) [inline src/biome/text/features.py:151]: See comment above

File: setup.py [hunk 3]
```diff
@@ -95,6 +95,7 @@ def about_info(package: str):
             "captum~=0.2.0",
             "ipywidgets~=7.5.1",
             "mlflow~=1.9.0",
+            "ray>=0.8.6,<0.9.0"
```
Comment (dcfidalgo) [inline setup.py:98]: they already released 1.0, can we use that?

**Review by dcfidalgo — COMMENTED** [review]
Thanks! Just a few inline comments.

File: src/biome/text/features.py (hunk 1 above)
Comment (frascuchon) [reply src/biome/text/features.py]: only keep `to_json` for allenlp `sanitize` integration

File: src/biome/text/features.py (hunk 2 above)
Comment (frascuchon) [reply src/biome/text/features.py]: same as before

File: setup.py (hunk 3 above)
Comment (frascuchon) [reply setup.py]: done

**Review by dcfidalgo — APPROVED** [review]

**Pull request merged**